### DISCUSSION TITLE
Multiple Power BI Dataset models in single report

### DISCUSSION THREAD

**Discussion by Declan1984** [TabularEditor/TabularEditor3#344] [Other Topics]
Hi - this is more broadly related to Power BI datasets, but as users of Tabular Editor, presumably most on here are working with those in the Power BI environment.

Is there currently any way to connect from a Power BI report to two or more Power BI datasets in the PBI service?  If not, any one know if this is even on the roadmap for Microsoft to implement?  As I'm thinking about how to further grow our data models, there are some core dataset models that have applicability across different use cases and groups.  Since it is not possible to have one single model to cover all potential scenarios and models, it would be great to have these core shareable datasets with standard measures that can be used in specific use cases alongside data related to the specific use case.

Otherwise, it seems like source data can be shared (via Data Flows) but the measures on those data in the form of dataset models need to be re-built in each use-case specific dataset model.

**Comment by Declan1984** [comment]
For reference, doing more research, have found that composite models are still in preview mode.  But this is basically the feature we need for model reusability.

**Reply by otykier** [reply]
This topic is also being discussed here: https://github.com/TabularEditor/TabularEditor3/issues/363

**Reply by Declan1984** [reply]
Encountered further issues with the test composite model I made where the measure definitions, even those that are defined in the composite dataset (not in the 'parent' datasets) are not accessible within TE3; the measure definition for the composite model measure is displayed with just a reference to the measure name and AS over DQ.  Measure could not be edited in TE3.
